test/unit/org/apache/cassandra/utils/RecomputingSupplierTest.java

```diff
@@ -93,31 +91,41 @@ public void recomputingSupplierTest() throws Throwable
                             lastSeen.accumulateAndGet(v, Math::max);
                             Assert.assertTrue(String.format("Seen %d out of order. Last seen value %d", v, seenBeforeGet),
                                               v >= seenBeforeGet);
-
                         }
-
                     }
                     catch (Throwable e)
                     {
                         thrown.set(e);
-                        latch.countDown();
                     }
                 }
             });
         }
 
-        latch.await(10, TimeUnit.SECONDS);
-
+        Util.spinAssertEquals(true, () -> counter.get() > 1000, 10);
```

Comment:
I think that the purpose of the latch was waiting for ten seconds to give time to the tasks to interact with the tested supplier, but abort immediately if an exception was thrown. Waiting for a fixed counter value of 1000 means a shorter wait in most of the cases, and less chances for races in both the test and the tested class.

Reply:
Would we prefer to give it 10s again then or is 1K iterations enough? :thinking: